lib/calendar_interval.ex

```diff
@@ -191,6 +191,8 @@ defmodule CalendarInterval do
 
   defp new(%NaiveDateTime{} = first, %NaiveDateTime{} = last, precision)
        when precision in @precisions do
+    # Ensure correct order of first/last date
+    [first, last] = Enum.sort([first, last], &lt?/2)
```

Comment:
I think the correct solution is to either:
1. have a new field on the struct: `direction :: :ascending | :descending`
2. crash

given, as you mentioned, other parts are making assumption about the order (and so does ISO 8601, kinda [1]) I think it's better to crash. 

[1] re iso 8601

I couldn't find anything specific preventing interval of `2018-01-31/2018-01-01`, what I found is the following:
- they used language like "start date", "end date", consecutive, successive, which sort of implies ascending ordering
- they allow to represent intervals as start date + duration, e.g.: `2018-01-01/P1W` (2018-01-01 + 1 week) and they specifically mention that duration can't be negative, so within that notation it's impossible to represent descending order
- new draft introduces notation e.g.: `2018-01-XX` (any day in january) which can be used in intervals: `2018-01-XX/2018-01-02`. If we allow descending order, `2018-01-XX/2018-01-15` is ambiguous, we don't know if it's ascending or descending
I'd say iso probably doesn't support descending order

Reply:
I like the idea of crashing on "invalid" ISO string intervals as well. If later there would be different ways to create an interval (e.g. the current defp new/3 becoming public) I'd still do the sorting because no matter the order – two dates always form an interval.